avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -1445,7 +1446,7 @@ private Schema toConnectSchema(org.apache.avro.Schema schema, boolean forceOptio
       while (paramIt.hasNext()) {
         Map.Entry<String, JsonNode> field = paramIt.next();
         JsonNode jsonValue = field.getValue();
-        if (!jsonValue.isTextual()) {
+        if ((!field.getKey().endsWith(DOC_PROP) && !jsonValue.isTextual())) {
```

Comment:
Doc fields are optional and can be null, ignore "*.doc" fields while reading connect parameters .

Reply:
@ewencp Sorry for bringing this up again. It would be great to see this specific issue fixed (and broaden the check later, in case more issues come up).